app.js

```diff
@@ -2,12 +2,27 @@ const express = require("express")
 const app = express()
 const {
     getTopics, 
-    getEndpoints
+    getEndpoints,
+    getArticleById
 } = require("./controllers/controller")
 
 app.get("/api/topics", getTopics)
+
 app.get("/api/", getEndpoints)
 
+app.get("/api/articles/:article_id", getArticleById)
+
+app.use((err, req, res, next) => {
+    if(err.status && err.msg) {
+        res.status(err.status).send({message: err.msg})
+    } 
+    if(err.code === '22P02') {
+        res.status(400).send({ message: "invalid article_id, bad request" });
+    }
+
```

Comment:
Great use of error handling middleware, it would be great if you could split these out into one for custom errors and one for PSQL errors, and even better if they were in a separate error file 
Check out the notes for examples https://notes.northcoders.com/courses/js-back-end/error-handling#4-refactoring-to-individual-error-handling-blocks